### ISSUE TITLE
helm lint documentation misleading

### ISSUE THREAD

**Issue by howardjohn** [helm/helm#6057]
Output of `helm version`:
```
$ helm version --client
Client: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
```

`helm lint --help` shows this documentation:
```
      --strict                   fail on lint warnings
```

However, this seems wrong. When I run `helm lint ...`, I get no output. When I run with `helm lint --strict ...` I get error messages.

My expectation is that I get warnings, an that if I enable `--strict` those result in an exit code of 1. But the warnings should always show up!.

Demo with https://github.com/istio/installer:

```
$ helm lint istio-control/istio-discovery -f global.yaml
==> Linting istio-control/istio-discovery
Lint OK

1 chart(s) linted, no failures
$ helm lint istio-control/istio-discovery -f global.yaml --strict
==> Linting istio-control/istio-discovery
[ERROR] templates/: render error in "istio-discovery/templates/deployment.yaml": template: istio-discovery/templates/deployment.yaml:15:32: executing "istio-discovery/templates/deployment.yaml" at <include (print $.Tem...>: error calling include: template: istio-discovery/templates/configmap.yaml:52:18: executing "istio-discovery/templates/configmap.yaml" at <.Values.pilot.authPo...>: map has no entry for key "authPolicy"

Error: 1 chart(s) linted, 1 chart(s) failed
```

**Comment by howardjohn** [comment]
Same with v2.14.2

**Comment by hickeyma** [comment]
I think this is related to #5992 and overall issues with linting.

**Comment by Aisuko** [comment]
I hit this issue on the helm v3 which I make a build from source code.

**Comment by technosophos** [comment]
I have found the issue that causes this. `--strict` _also_ turns on extra flags in the template engine which will report errors if a template value evaluates to `nil`. 

There is a PR here to change that behavior. Please weigh in if you have thoughts about whether you would like to see this behavior changed: #8017

**Comment by technosophos** [comment]
Closed by #8017
